### PR TITLE
chore: add tests to api V1 - part 2

### PR DESCRIPTION
**Pull request by victorvhs017**
## What does this PR do?

Add tests to all file in the API v1 route, except for types and route.ts files;

## How should this be tested?

Run the test command in the web project

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

## Summary by CodeRabbit

- **Tests**
  - Added comprehensive new test suites covering user session retrieval, contact retrieval, survey response creation and retrieval, signed URL generation, survey deletion, survey listing, webhook deletion, and webhook creation.
  - Improved test coverage for error handling, validation, caching, and edge cases across various API functions.
  - Minor cleanup in segment settings component tests by removing a commented line.
  - Adjusted timing assertion in delay function test to allow slight execution variability.

### CODE REVIEW

File: apps/web/app/api/v1/management/responses/lib/response.test.ts [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    test("should handle generic errors", async () => {
+      const genericError = new Error("Something went wrong");
+      vi.mocked(prisma.response.findMany).mockRejectedValue(genericError);
+
+      await expect(getResponsesByEnvironmentIds(mockEnvironmentIds)).rejects.toThrow(genericError);
+      expect(cache).toHaveBeenCalled();
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (victorvhs017) [reply apps/web/app/api/v1/management/responses/lib/response.test.ts:347]: We'll add it later

**Review by gupta-piyush19 — APPROVED** [review]
LGTM 🚀
